sound/soc/sof/ipc.c

```diff
@@ -661,9 +661,16 @@ int snd_sof_ipc_set_get_comp_data(struct snd_sof_control *scontrol,
 	struct sof_ipc_fw_ready *ready = &sdev->fw_ready;
 	struct sof_ipc_fw_version *v = &ready->version;
 	struct sof_ipc_ctrl_data_params sparams;
+	struct snd_sof_widget *swidget;
 	size_t send_bytes;
 	int err;
 
+	swidget = sof_find_widget_by_comp_id(sdev, scontrol->comp_id);
+
+	/* return if the widget has not been set up */
+	if (!swidget->ref_count)
+		return 0;
```

Comment:
just to make sure we have thought about everything: for reading returning 0 here won't confuse the caller into thinking, that they have now valid data from the widget? E.g. during pre-caching?

Reply:
a normal get() would work OK because we always cache the data at the end of topology loading. But for volatile_get() this is indeed misleading. I've changed this to return 0 for SET commands and -EIO for GET commands. ONly volatile gets would even get to this point and they would require the widget to be set up to perform a GET